src/controllers/queryRunner.ts

```diff
@@ -679,6 +660,74 @@ export default class QueryRunner {
         }
         await p;
 
+        this.constructCopyString(
+            copyString,
+            rowIdToRowMap,
+            rowIdToSelectionMap,
+        );
+
+        await this.writeStringToClipboard(copyString);
+    }
+
+    public async exportCellsToClipboard(
+        data: DbCellValue[][],
+        batchId: number,
+        resultId: number,
+        selection: ISlickRange[],
+        headersFlag,
+    ) {
+        let copyString = "";
+        if (headersFlag) {
+            copyString = this.addHeadersToCopyString(
+                copyString,
+                batchId,
+                resultId,
+                selection,
+            );
+        }
+
+        // create a mapping of rows to selections
+        let rowIdToSelectionMap = new Map<number, ISlickRange[]>();
+        let rowIdToRowMap = new Map<number, DbCellValue[]>();
+
+        // create a mapping of the ranges to get promises
+        let tasks = selection.map((range) => {
+            return async () => {
+                const result = data;
+                let count = 0;
+                for (let row of result) {
+                    let rowNumber = count + range.fromRow;
+                    if (rowIdToSelectionMap.has(rowNumber)) {
+                        let rowSelection = rowIdToSelectionMap.get(rowNumber);
+                        rowSelection.push(range);
+                    } else {
+                        rowIdToSelectionMap.set(rowNumber, [range]);
+                    }
+                    rowIdToRowMap.set(rowNumber, row);
+                    count += 1;
+                }
+            };
+        });
```

Comment:
Can you add more comments here?

* what do the promises represent?
* what are the rowId mappings being used to calculate?